src/map/battle.cpp

```diff
@@ -1395,6 +1472,21 @@ bool battle_status_block_damage(struct block_list *src, struct block_list *targe
 			status_change_end(target, SC_BUNSINJYUTSU);
 		return false;
 	}
+	
+	if ((sce = sc->getSCE(SC_DAMAGE_HEAL))) {
+		if (damage > 0 && (flag & sce->val2)) {
+			int heal = min(static_cast<int>(damage), INT32_MAX );
```

Comment:
```suggestion
			int32 heal = static_cast<int32>( min( damage, INT32_MAX ) );
```
If you cast it too early it maybe is getting turned into a negative number.